### PR TITLE
When an image path is hovered always show the image from its uri.

### LINKED ISSUE #101: Request textDocument/hover failed on image.mimeType
VS Code 1.21.1, Plugin 2.1.10

1. Open this asset.
```json
{
    "asset": {
        "version": "2.0"
    },
    "images": [
        {
            "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAADElEQVQYV2P4z/AfAAQAAf+I0P3MAAAAAElFTkSuQmCC",
            "mimeType": "image/png"
        }
    ]
}
```

2. Hover over `image/png`.
3. VS Code output:
```
Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: Cannot read property 'startsWith' of undefined
  Code: -32603 
```

### PR DESCRIPTION
**Pull request by najadojo**
fixes #101

### CODE REVIEW

File: server/src/server.ts [hunk 1]
```diff
@@ -522,11 +522,13 @@ connection.onHover((textDocumentPosition: TextDocumentPositionParams): Hover =>
...
+        let splitPath = path.split('/');
+        if (splitPath.length > 3) {
+            path = splitPath.slice(0, 3).join('/');
+        }
         let imageData = getFromPath(pathData.jsonMap.data, path);
         if (imageData) {
-            if (!path.endsWith('/uri')) {
-                imageData = imageData.uri;
-            }
+            imageData = imageData.uri;
```
Comment (emackey) [inline server/src/server.ts]: This can crash when `uri` is left unspecified, right?  Granted the image is not much use without a uri.

Comment (najadojo) [reply server/src/server.ts]: It looks like the validator has an error if there isn't a uri too.

Comment (emackey) [reply server/src/server.ts]: I see that too.  @lexaknyazev Do you need a new issue for that one?

**Comment by emackey** [comment]
Thanks @najadojo!

File: server/src/server.ts (hunk 1 above)
Comment (lexaknyazev) [reply server/src/server.ts]: Validator has been fixed in `2.0.0-dev.2.0`.

Comment (emackey) [reply server/src/server.ts]: Aha, I had just un-done the validator upgrade to test this PR.  I should have tested the two together!

**Pull request merged**